jOOQ-codegen/src/main/java/org/jooq/util/JavaGenerator.java

```diff
@@ -278,6 +279,9 @@ public final void generate(Database db) {
             + ((!generateGeneratedAnnotation && (useSchemaVersionProvider || useCatalogVersionProvider)) ?
                 " (forced to true because of <schemaVersionProvider/> or <catalogVersionProvider/>)" : ""));
         log.info("  JPA annotations", generateJPAAnnotations());
+        log.info("  JPA version", generateJpaVersion()
+            + (!generateJPAAnnotations && generateJpaVersion != null ? " (forced to null because of <jpaAnnotations/>)"  :
+              (generateJPAAnnotations && generateJpaVersion == null ? " (forced to " + JpaVersion.latest().name() + " because of <jpaVersion/>)" : "")));
```

Comment:
I think in this case, it's sufficient to simply print the version itself. No need to add any logic (given that we won't have the enum in Java)